Core/Src/stateMachine.c

```diff
@@ -251,80 +268,100 @@ uint32_t sm_fault_return(acc_data_t *accData)
 		fault_table[6].data_1 = fault_data->min_voltage.val;
 	}
 
-	static uint32_t fault_status = 0;
 	int incr = 0;
 	while (fault_table[incr].id != NULL) {
-		fault_status |= sm_fault_eval(&fault_table[incr]);
+		uint32_t item_code = fault_table[incr].code;
+		if (sm_fault_eval(&fault_table[incr])) {
+			if (fault_table[incr].is_critical) {
+				fault_status_crit |= item_code;
+			} else {
+				fault_status_noncrit |= item_code;
+			}
+		} else {
+			// Clear bit for non-critical faults
+			if (!fault_table[incr].is_critical) {
+				fault_status_noncrit &= ~item_code;
+			}
+		}
 		incr++;
 	}
-	// TODO: Remove This !!!!
-	fault_status &= ~DISCHARGE_LIMIT_ENFORCEMENT_FAULT;
-	return fault_status;
+
+	// TODO: Remove This !!!! (because this is actually a non-critical fault?)
+	// fault_status &= ~DISCHARGE_LIMIT_ENFORCEMENT_FAULT;
```

Comment:
@Sabramz hmmm should this be here or not

Reply:
I do not know. Something tells me this was a debugging thing done in a rush last year. Lets bring DCL fault back in, and if it gives us problems we can remove it easily.